test/module/irohad/pending_txs_storage/pending_txs_storage_test.cpp

```diff
@@ -130,6 +129,94 @@ TEST_F(PendingTxsStorageFixture, InsertionTest) {
+  storage_->updatedBatchesHandler(state);
+  const auto &creator = "alice@iroha";
+  auto pending = storage_->getPendingTransactions(creator, kPageSize, std::nullopt, first_time, last_time);
+  IROHA_ASSERT_RESULT_VALUE(pending);
+  ASSERT_EQ(expected.all_transactions_size, pending.assumeValue().transactions.size() );
+}
+
+/**
+ * PaginationMeta works in PendingTxsStorage
+ * @given Batch of two transactions and storage
+ * @when storage receives updated mst state with the batch
+ * @then list of pending transactions starting from specified timestamp can be obtained
+ */
+TEST_F(PendingTxsStorageFixture, TxPaginationTestFirstTimeSpecified) {
+  auto state = emptyState();
+  auto transactions = twoTransactionsBatch();
+  *state += transactions;
+  auto first_time = getUniqueTime();
+  auto transactions1 = twoTransactionsBatch();
+  *state += transactions1;
+  const auto kPageSize = 100u;
+  Response expected;
+  expected.transactions.insert(expected.transactions.end(),
+                               transactions1->transactions().begin(),
+                               transactions1->transactions().end());
+  expected.all_transactions_size = transactions1->transactions().size();
+  storage_->updatedBatchesHandler(state);
+  const auto &creator = "alice@iroha";
+  auto pending = storage_->getPendingTransactions(
+      creator, kPageSize, std::nullopt, first_time);
+  IROHA_ASSERT_RESULT_VALUE(pending);
+  ASSERT_EQ(expected.all_transactions_size,
+            pending.assumeValue().transactions.size());
+}
+
+/**
+ * PaginationMeta works in PendingTxsStorage
+ * @given Batch of two transactions and storage
+ * @when storage receives updated mst state with the batch
+ * @then list of pending transactions up to specified timestamp can be
+ * obtained
+ */
+TEST_F(PendingTxsStorageFixture, TxPaginationTestLastTimeSpecified) {
+  auto state = emptyState();
+  auto transactions = twoTransactionsBatch();
+  *state += transactions;
+  auto transactions1 = twoTransactionsBatch();
+  *state += transactions1;
+  auto last_time = getUniqueTime();
+  const auto kPageSize = 100u;
+  Response expected;
+  expected.transactions.insert(expected.transactions.end(),
+                               transactions->transactions().begin(),
+                               transactions->transactions().end());
+  expected.transactions.insert(expected.transactions.end(),
+                               transactions1->transactions().begin(),
+                               transactions1->transactions().end());
+  expected.all_transactions_size =
+      transactions->transactions().size() + transactions1->transactions().size();
+  storage_->updatedBatchesHandler(state);
+  const auto &creator = "alice@iroha";
+  auto pending = storage_->getPendingTransactions(
+      creator, kPageSize, std::nullopt, std::nullopt, last_time);
+  IROHA_ASSERT_RESULT_VALUE(pending);
+  ASSERT_EQ(expected.all_transactions_size,
+            pending.assumeValue().transactions.size());
+}
+
```

Comment:
I suggest also failure tests:
1. First timestamp after last transaction.
2. First timestamp after last timestamp

Reply:
FirstTimestamp after last timestamp is already checked here:  https://github.com/Pawlak00/iroha/blob/f8dc0d3466c2b60f4a5bdb8c1e2750b82b82328f/test/module/shared_model/validators/protobuf/proto_query_validator_test.cpp and validation of whole TxPaginationMeta is independent from query I think